docs: Update architecture reference page (#1275)
canonical/k8s-snap: docs/canonicalk8s/snap/reference/architecture.md

![cluster5][]

Actors that interact with the K8s snap:

- **K8s admin**:  interacts directly with the Kubernetes API server. {{product}}
provides out of the box admin access to configure the cluster.
- **K8s user** : consumes the workloads hosted in the cluster.

Non-human users of the K8s snap:

- [`k8s-operator charm`][K8s charm]: uses the snap to drive the Kubernetes
cluster and orchestrate multi-node clustering operations.

Although {{product}} provides its own implementation of the following services,
external systems can be easily integrated:

- **Load Balancer**
- **Storage**
- **Identity management**
- **External datastore**

## The k8s snap

Contained within the K8s snap itself:

![cluster1][]

The `k8s` snap distribution includes the following:

- **Kubectl**: through which users and other systems interact with Kubernetes
and drive the cluster operations.

> - **K8s control plane**: the Kubernetes services, as well as core
> workloads built from upstream and shipped in the snap.

Reviewer: In the upstream docs, they refer to them as [`core components`](https://kubernetes.io/docs/concepts/overview/components/#core-components). Maybe we could align our definitions with that?
```suggestion
- **K8s core components**: the Kubernetes services, as well as core
workloads built from upstream and shipped in the snap.
```

Author: Do you think we should also update the diagrams?

Reviewer: Sorry @nhennigan! I missed the diagrams, they are hard to see in the Github pane. But yes, let's change that to K8s core components and we can reuse the same description from the list.

- **Kubernetes datastore**: uses [Dqlite] to store data on the state of the
cluster. It can be replaced by an external datastore.
- **Cluster datastore**: uses Dqlite as a replicated database to store cluster
configuration.

> - **Runtime**: `containerd` is the shipped container runtime.

Reviewer: ```suggestion
- **Container runtime**: `containerd` is the shipped container runtime.
```

![cluster5][]

Actors that interact with the K8s snap:

- **K8s admin**:  interacts directly with the Kubernetes API server. {{product}}
provides out of the box admin access to configure the cluster.
- **K8s user** : consumes the workloads hosted in the cluster.

Non-human users of the K8s snap:

- [`k8s-operator charm`][K8s charm]: drives the Kubernetes cluster and
orchestrates multi-node clustering operations.

Although {{product}} provides its own implementation of the following services,
external systems that can be easily integrated:

- **Load Balancer**
- **Storage**
- **Identity management**
- **External datastore**

## The k8s snap

Contained within the K8s snap itself:

![cluster1][]

The `k8s` snap distribution includes the following:

- **Kubectl**: through which users and other systems interact with Kubernetes
and drive the cluster operations.
- **K8s control plane**: the Kubernetes services, as well as core
workloads built from upstream and shipped in the snap.

> - **Kubernetes datastore**: uses dqlite to store data on the state of the
> cluster. It can be replaced by an external datastore.
> - **Cluster datastore**: uses dqlite as a replicated database to store cluster

Reviewer 2: ```suggestion
- **Kubernetes datastore**: uses Dqlite to store data on the state of the
cluster. It can be replaced by an external datastore.
- **Cluster datastore**: uses Dqlite as a replicated database to store cluster
```
Capitalized?

## Canonical K8s charms

Canonical `k8s` charms are the [`k8s` charm][K8s
charm] and the [`k8s-worker` charm][K8s-worker charm].

![cluster4][]

The {{product}} charms include the following:

> - **`k8s`** : installs and manages the `k8s` snap on control plane node. Also
> supports integration with other compatible charms.

Reviewer 2: ```suggestion
- **`k8s`** : installs and manages the `k8s` snap on control plane nodes. The charm also
supports integrations with other compatible charms.
```

> - **`k8s-worker`**: installs and manages the `k8s` snap on worker node.

Reviewer 2: ```suggestion
- **`k8s-worker`**: installs and manages the `k8s` snap on worker nodes.
```
